pythran/pythonic/types/bool.hpp

```diff
@@ -3,4 +3,35 @@
 
 #include "pythonic/include/types/bool.hpp"
 
+#ifdef ENABLE_PYTHON_MODULE
+
+#include "pythonic/python/core.hpp"
+
+namespace pythonic
+{
+  template <>
+  struct to_python<bool> {
+    static PyObject *convert(bool b)
+    {
+      if (b)
+        Py_RETURN_TRUE;
+      else
+        Py_RETURN_FALSE;
+    }
+  };
+  template <>
+  struct from_python<bool> {
+    static bool is_convertible(PyObject *obj)
+    {
+      return PyBool_Check(obj);
+    }
+    static bool convert(PyObject *obj)
+    {
+      return PyInt_AsLong(obj);
+    }
+  };
+}
+
+#endif
+
```

Comment:
For namespace impl in header, It make sens to me when it is type related stuff (like return type for enumerate or other iterators) but I agree that others should not be in include files.

Also, I don't really like the idea of having a correct include order. We should include the dependencies in the file but not having an include order as it is the issue I tried to fix with the include split stuff. If it is not possible, I think the include declaration will have to be done (to_python and from_python are part of the interface as a converter interface. It is called from outside the class.)